### PR TITLE
Sw dspdc 1064 codegen init method

### PR DESCRIPTION
**Pull request by snwessel**
Adding an init method to the codegen in MonsterJadeDatasetPlugin. ([Jira Ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1064))

Also adding a few more helper methods to avoid duplicate code.

One question: Is there anything I need to do to bump the version?

### CODE REVIEW

File: plugins/jade/src/test/scala/org/broadinstitute/monster/sbt/ClassGeneratorSpec.scala [hunk 1]
```diff
@@ -354,6 +419,11 @@ class ClassGeneratorSpec extends AnyFlatSpec with Matchers with EitherValues {
        |      _root_.io.circe.derivation.renaming.snakeCase,
        |      _root_.scala.None
        |    )
+       |
+       |  def init(): StructColumn = {
+       |    StructColumn(
+       |      comment = _root_.scala.Option.empty[_root_.$structPackage.RowComment])
```
Comment (danxmoran) [inline plugins/jade/src/test/scala/org/broadinstitute/monster/sbt/ClassGeneratorSpec.scala:425]: Looks like a bug here, `$structPackage` not being substituted properly.

Comment (danxmoran) [reply plugins/jade/src/test/scala/org/broadinstitute/monster/sbt/ClassGeneratorSpec.scala:425]: Unless we're confusingly using the literal string "$structPackage" in the tests...

Comment (snwessel) [reply plugins/jade/src/test/scala/org/broadinstitute/monster/sbt/ClassGeneratorSpec.scala:425]: I think it is substituting out the `$structPackage` value with "rab.oof" when it actually runs the tests. I just re-ran the test and had it print the actual output, which looks like this: 
```
case class StructColumn(
comment: _root_.scala.Option[_root_.rab.oof.RowComment])

object StructColumn {de / Test / executeTests 1s
  implicit val encoder: _root_.io.circe.Encoder[StructColumn] =
    _root_.io.circe.derivation.deriveEncoder(
      _root_.io.circe.derivation.renaming.snakeCase,
      _root_.scala.None
    )

  def init(): StructColumn = {
    StructColumn(
      comment = _root_.scala.Option.empty[_root_.rab.oof.RowComment])
  }
}
```

Comment (danxmoran) [reply plugins/jade/src/test/scala/org/broadinstitute/monster/sbt/ClassGeneratorSpec.scala:425]: Oh! I missed the `s` at the start of each string.

Comment (snwessel) [reply plugins/jade/src/test/scala/org/broadinstitute/monster/sbt/ClassGeneratorSpec.scala:425]: They are defined above as:
```
  private val testPackage = "foo.bar"
  private val structPackage = testPackage.reverse
```

**Comment by danxmoran** [comment]
Re: Bumping the version, this project uses git tags to drive the process. After the PR merges you can do:
```
git fetch --tags
git tag -l
```
To see the latest version. Then you:
```
git tag v<next-logical-version>
git push origin v<newly-created-version>
```

**Pull request merged**